### PR TITLE
Add TOC.

### PR DESCRIPTION
**Pull request by jorgebucaran**
This PR adds a Table of Contents to the README. Sometimes I think it's useful to have one, but I also like to keep things simple, so I am not sure I want to include this or not.

### CODE REVIEW

**Comment by okwolf** [comment]
Since we've consolidated the docs down to a single README that has about 15 screenfuls of content, it might be nice to provide these for jumping around on that page.

File: README.md [hunk 1]
```diff
@@ -8,6 +8,37 @@ Hyperapp is a JavaScript library for building web applications.
 * **Functional** — Hyperapp's design is inspired by [The Elm Architecture](https://guide.elm-lang.org/architecture). Create scalable browser-based applications using a functional paradigm. The twist is you don't have to learn a new language.
 * **Batteries-included** — Out of the box, Hyperapp combines state management with a Virtual DOM engine that supports keyed updates & lifecycle events — all with no dependencies.
 
+<h2> Table of Contents</h2>
+<!-- TOC -->
```
Comment (Swizz) [inline README.md]: What about the use of `details` and `summary` ? 
```
<details>
  <summary>Table of Contents</summary>
</details>
```

**Comment by selfup** [comment]
Add it! README is large enough to justify it 🎉

**Comment by lukejacksonn** [comment]
Feels like there is consensus here to go ahead and add the TOC. It can always be removed again if we don't like it.. right? 😅

**Comment by okwolf** [comment]
We should merge or close this. No point in waiting forever.

**Comment by jorgebucaran** [comment]
Not sure yet! 🙏

**Comment by jorgebucaran** [comment]
I know this change makes the README easier to use, but the docs are so minimal one could read the whole thing in a seat if sufficiently motivated. The truth is I don't want to merge this because I feel it makes the documentation look big and scary. I tried some variations of the ToC, but I couldn't find satisfaction in any of them.

On the sidebar of a website, this would be a different story.

**Pull request closed without merging**